statsmodels/tsa/statespace/tests/test_sarimax.py

```diff
@@ -0,0 +1,1149 @@
+        )
+
+        params = np.r_[true['params_exog'], true['params_ar'],
+                       true['params_ma'], true['params_variance']]
+
+        self.update(params)
+
+
+class TestFriedmanMLERegression(Friedman):
+    def __init__(self):
+        super(TestFriedmanMLERegression, self).__init__(
+            results_sarimax.friedman2_mle
+        )
+        self.result = self.filter()
+
+
+class TestFriedmanStateRegression(Friedman):
+    def __init__(self):
+        # Remove the regression coefficients from the parameters, since they
+        # will be estimated as part of the state vector
+        true = dict(results_sarimax.friedman2_mle)
+
+        true['mle_params_exog'] = true['params_exog'][:]
+        true['mle_se_exog'] = true['se_exog'][:]
+
+        true['params_exog'] = []
+        true['se_exog'] = []
+
+        super(TestFriedmanStateRegression, self).__init__(
+            true, mle_regression=False
+        )
+
+        self.result = self.filter()
+
+    def test_regression_parameters(self):
+        assert_almost_equal(
+            self.result.filtered_state[-2:, -1],
+            self.true['mle_params_exog'], 1
+        )
+
+    # Loglikelihood (and so aic, bic) is slightly different when states are
+    # integrated into the state vector
+    def test_loglike(self):
+        pass
+
+    def test_aic(self):
+        pass
+
+    def test_bic(self):
+        pass
+
+
+class TestFriedmanPredict(Friedman):
+    """
+    ARMAX model: Friedman quantity theory of money, prediction
+
+    Stata arima postestimation documentation, Example 1 - Dynamic forecasts
+
+    This follows the given Stata example, although it is not truly forecasting
+    because it compares using the actual data (which is available in the
+    example but just not used in the parameter MLE estimation) against dynamic
+    prediction of that data. Here `test_predict` matches the first case, and
+    `test_dynamic_predict` matches the second.
+    """
+    def __init__(self):
+        super(TestFriedmanPredict, self).__init__(
+            results_sarimax.friedman2_predict
+        )
+
+        self.result = self.filter()
+
+    # loglike, aic, bic are not the point of this test (they could pass, but we
+    # would have to modify the data so that they were calculated to
+    # exclude the last 15 observations)
+    def test_loglike(self):
+        pass
+
+    def test_aic(self):
+        pass
+
+    def test_bic(self):
+        pass
+
+    def test_predict(self):
+        assert_almost_equal(
+            self.result.predict()[0],
+            self.true['predict'], 3
+        )
+
+    def test_dynamic_predict(self):
+        dynamic = len(self.true['data']['consump'])-15-1
+        assert_almost_equal(
+            self.result.predict(dynamic=dynamic)[0],
+            self.true['dynamic_predict'], 3
+        )
+
+
+class TestFriedmanForecast(Friedman):
+    """
+    ARMAX model: Friedman quantity theory of money, forecasts
```

Comment:
Docstrings in test classes do not result in that behavior for nosetests.

We talked about this in one of the comments above (from the summer) and at that point decided to leave it with the docstrings, but I can convert them to comments if that's preferred.